src/components/table/SnapshotAccessRequestTable.tsx

```diff
@@ -55,16 +92,29 @@ function SnapshotAccessRequestTable({
       ) * -1,
   );
 
+  const openDetailsModal = (request: SnapshotAccessRequestResponse) => {
+    setSelectedAccessRequest(request);
+  };
+
+  useEffect(() => {
+    selectedAccessRequest?.id !== undefined &&
+      dispatch(getSnapshotAccessRequestDetails(selectedAccessRequest.id));
+  }, [dispatch, selectedAccessRequest]);
+
   const columns: Array<TableColumnType> = [
     {
       label: 'Request Name',
       name: 'snapshotName',
       render: (row: SnapshotAccessRequestResponse) => (
-        <TextWithModalDetails
-          modalText={row.snapshotName}
-          modalHeading={`Summary for: ${row.snapshotName}`}
-          modalContent={row.summary}
-        />
+        <Button
+          className={classes.openButton}
+          aria-label={row.snapshotName}
+          onClick={() => openDetailsModal(row)}
+          disableFocusRipple={true}
+          disableRipple={true}
+        >
+          {row.snapshotName}
+        </Button>
```

Comment:
why change to a button here? How was the modal working before?

Reply:
Previously the button was within the modal - that worked because there was no additional API call. There is now, so we want to manage the async behavior separately